src/ur/ur_driver.cpp

```diff
@@ -583,6 +589,61 @@ std::vector<std::string> UrDriver::getRTDEOutputRecipe()
 
 void UrDriver::setKeepaliveCount(const uint32_t& count)
 {
+  URCL_LOG_ERROR("DEPRECATION NOTICE: Setting the keepalive count has been deprecated. Instead use the "
+                 "watchdog, to set the timeout directly in the write commands. Please change your code to set the "
+                 "read timeout in the write commands directly. This keepalive count will overwrite the timeout passed "
+                 "to the write functions.");
   reverse_interface_->setKeepaliveCount(count);
 }
+
+float UrDriver::verifyWatchdogTimeout(const Watchdog& watchdog, const comm::ControlMode& control_mode,
+                                      const float& step_time)
+{
+  if (comm::ControlModeTypes::is_control_mode_non_realtime(control_mode))
+  {
+    if (watchdog.timeout_ < step_time && watchdog.timeout_ > 0.0)
+    {
+      std::stringstream ss;
+      ss << "Watchdog timeout " << watchdog.timeout_ << " is below the step time " << step_time
+         << ". It will be reset to the step time.";
+      URCL_LOG_ERROR(ss.str().c_str());
+      return step_time;
+    }
+    else
+    {
+      return watchdog.timeout_;
+    }
+  }
+  else if (comm::ControlModeTypes::is_control_mode_realtime(control_mode))
+  {
+    float max_realtime_timeout = 1.0;
```

Comment:
While I agree with a mechanism to prevent users from having large timeouts on rt-commanding setting this here to 1 second seems a bit arbitrary. However, that's just a minor comment.

Reply:
Yes it is an arbitrary timeout and I don't have a good argument for the 1 second, other than I don't think we should make it to large. So I am open for suggestions.